ociclient/cache/cache.go

```diff
@@ -46,11 +50,36 @@ func NewCache(log logr.Logger, options ...Option) (Cache, error) {
 		overlay = memoryfs.New()
 	}
 
+	//initialize metrics
+	metrics.CachedItems.WithLabelValues(opts.BasePath).Set(0)
+	metrics.CacheDiskUsage.WithLabelValues(opts.BasePath).Set(0)
+	metrics.CacheHits.WithLabelValues(opts.BasePath).Add(0)
+	if opts.InMemoryOverlay {
+		metrics.CacheMemoryUsage.WithLabelValues(opts.BasePath).Set(0)
+	}
+
+	err = vfs.Walk(base, "/", func(_ string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			metrics.CachedItems.WithLabelValues(opts.BasePath).Inc()
+			metrics.CacheDiskUsage.WithLabelValues(opts.BasePath).Add(float64(info.Size()))
+		}
+		return nil
+	})
+
+	if err != nil {
+		log.V(7).Info("Error during initialization of cache metrics.")
+		return nil, err
```

Comment:
I think we should not log that error as a info.
I would let the caller of the function decide whether he wants to log the error.
```suggestion
		return nil, fmt.Errorf("error during initialization of cache metrics: %w", err)
```